scraper.ts

```diff
@@ -0,0 +1,241 @@
+// Reads and parses anchor data files from bikeshed-data repository and writes:
+// - xref.json containing parsed and formatted data by term
+// - specs.json having data by spec shortname
+// - specmap.json having spec details
+
+import { promises as fs, existsSync } from 'fs';
+import { resolve as resolvePath, join as joinPath } from 'path';
+import { spawn } from 'child_process';
+import Trie from 'compact-prefix-tree/index.js';
+import { SUPPORTED_TYPES, DATA_DIR } from './constants.js';
+import { Data } from './search.js';
+
+const { readdir, readFile, writeFile } = fs;
+
+const INPUT_DIR_BASE = joinPath(DATA_DIR, 'bikeshed-data', 'data');
+const SPECS_JSON = resolvePath(INPUT_DIR_BASE, './specs.json');
+const INPUT_ANCHORS_DIR = resolvePath(INPUT_DIR_BASE, './anchors/');
+
+const OUT_DIR_BASE = joinPath(DATA_DIR, 'xref');
+const OUTFILE_BY_TERM = resolvePath(OUT_DIR_BASE, './xref.json');
+const OUTFILE_BY_SPEC = resolvePath(OUT_DIR_BASE, './specs.json');
+const OUTFILE_SPECMAP = resolvePath(OUT_DIR_BASE, './specmap.json');
+
+type ParsedDataEntry = ReturnType<typeof parseData>[0];
+
+interface DataByTerm {
+  [term: string]: Omit<ParsedDataEntry, 'key' | 'isExported'>[];
+}
+interface DataBySpec {
+  [shortname: string]: Omit<ParsedDataEntry, 'shortname' | 'isExported'>[];
+}
+
+export async function main() {
+  const hasUpdated = await updateInputSource();
+  if (!hasUpdated) {
+    console.log('Nothing to update');
```

Comment:
Might be good to put the module name here. Otherwise it’s not clear who put this in the log. Same with the other file I just reviewed.